jbpm-flow/src/main/java/org/jbpm/process/instance/context/variable/VariableScopeInstance.java

```diff
@@ -173,6 +173,7 @@ public void enforceRequiredVariables() {
         VariableScope variableScope = getVariableScope();
         for (Variable variable : variableScope.getVariables()) {
             String name = variable.getName();
+			Object defaultValue = variable.getMetaData("defaultValue");
```

Comment:
this will cause the check to be triggered before tha default value is set, therefore causing an error when default value is provided (which is not coherent)